### PR TITLE
Session and entity dirty check implementation

### PR DESCRIPTION
**Pull request by DanielChanger**

### CODE REVIEW

File: src/main/java/com/example/Session.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            try (PreparedStatement preparedStatement = connection.prepareStatement(updateByIdQuery)) {
+                preparedStatement.setObject(1, getIdField(entityType).get(entityInstance));
+                preparedStatement.executeUpdate();
+            }
+        }
+    }
+
+    private static String mapToSetClauseString(Map<String, Object> columnsToUpdate) {
+        return columnsToUpdate.entrySet()
+                .stream().map(e -> "%s = %s" .formatted(e.getKey(), e.getValue()))
```
Comment (StanislavZabramnyi) [inline src/main/java/com/example/Session.java:134]: Typo here, the new value of column should be wrapped by `''` like `name = 'Simba'`

File: src/main/java/com/example/Session.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    @SneakyThrows
+    private void update(Object entityInstance, Map<String, Object> columnsToUpdate) {
+        try (Connection connection = dataSource.getConnection()) {
+            Class<?> entityType = entityInstance.getClass();
+            String tableName = getTableName(entityType);
+            String idColumnName = getColumnName(getIdField(entityType));
+            String setClauseString = mapToSetClauseString(columnsToUpdate);
+            String updateByIdQuery = "UPDATE %s SET %s WHERE %s = " .formatted(tableName, setClauseString, idColumnName);
```
Comment (StanislavZabramnyi) [inline src/main/java/com/example/Session.java:124]: The `?` is missed in the end of query

File: src/main/java/com/example/Session.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                return entity;
+            }
+        }
+    }
+
+    @SneakyThrows
+    private static <T> Map<String, Object> createFieldNameToValueMap(T entity) {
+        Map<String, Object> map = new HashMap<>();
+        for (Field field : entity.getClass().getDeclaredFields()) {
+            Object value = field.get(entity);
```
Comment (StanislavZabramnyi) [inline src/main/java/com/example/Session.java:55]: Potential `IllegalAccessException` here, `field.setAccessible(true)` should be used for cases of private fields

File: src/main/java/com/example/Session.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    @SneakyThrows
+    private void update(Object entityInstance, Map<String, Object> columnsToUpdate) {
+        try (Connection connection = dataSource.getConnection()) {
+            Class<?> entityType = entityInstance.getClass();
+            String tableName = getTableName(entityType);
+            String idColumnName = getColumnName(getIdField(entityType));
+            String setClauseString = mapToSetClauseString(columnsToUpdate);
+            String updateByIdQuery = "UPDATE %s SET %s WHERE %s = " .formatted(tableName, setClauseString, idColumnName);
+            try (PreparedStatement preparedStatement = connection.prepareStatement(updateByIdQuery)) {
+                preparedStatement.setObject(1, getIdField(entityType).get(entityInstance));
```
Comment (StanislavZabramnyi) [inline src/main/java/com/example/Session.java:126]: Potential `IllegalAccessException` here, `.setAccessible(true)` should be used for cases of private fields

File: src/main/java/com/example/Session.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        for (Field field : entity.getClass().getDeclaredFields()) {
+            Object value = field.get(entity);
+            map.put(field.getName(), value);
+        }
+        return map;
+    }
+
+    @SneakyThrows
+    private <T> T mapResultSetToEntity(ResultSet resultSet, Class<T> entityType) {
+        Field[] declaredFields = entityType.getDeclaredFields();
```
Comment (StanislavZabramnyi) [inline src/main/java/com/example/Session.java:63]: Call `resultSet.next()` before getting objects from the resultSet, as a `ResultSet` cursor is initially positioned before the first row

File: src/main/java/com/example/model/Product.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+import java.math.BigDecimal;
+
+@Table("PRODUCT")
+@Setter
+@ToString
+public class Product {
+
+    @Id
+    @Column("ID")
```
Comment (szabramnyi) [inline src/main/java/com/example/model/Product.java:17]: Using capital letters to name database tables or columns is not a good practice

Comment (StanislavZabramnyi) [inline src/main/java/com/example/model/Product.java:17]: Using capital letters to name database tables or columns is not a good practice

**Pull request closed without merging**